### ISSUE TITLE
resourcequota scopes should support user-defined scopes

### ISSUE THREAD

**Issue by ddongchen** [kubernetes/kubernetes#61702]
> /kind feature
>/sig resource quota

 For the "ResourceQuota"->"ResourceQuotaSpec"->"Scopes", we just support defined scopes, they are:
const (
	// Match all pod objects where spec.activeDeadlineSeconds
	ResourceQuotaScopeTerminating ResourceQuotaScope = "Terminating"
	// Match all pod objects where !spec.activeDeadlineSeconds
	ResourceQuotaScopeNotTerminating ResourceQuotaScope = "NotTerminating"
	// Match all pod objects that have best effort quality of service
	ResourceQuotaScopeBestEffort ResourceQuotaScope = "BestEffort"
	// Match all pod objects that do not have best effort quality of service
	ResourceQuotaScopeNotBestEffort ResourceQuotaScope = "NotBestEffort"
)

  Shall we support user-defined scopes? For example, I created different resource quota for different workload in the same namespace, and I want to check the separated quota of different workload.

**Comment by lavalamp** [comment]
cc @derekwaynecarr 
cc @yliaog

**Comment by yliaog** [comment]
supposedly workload in one namespace are started by one team, how the quota allocated to the team is sub-allocated to each workload is managed by the team, providing a finer granularity of quota management within a namespace is not very useful, IMO.

**Comment by ddongchen** [comment]
Yes, for now it's may be little useful. While i think it's better to make the resource quota scopes more flexible,  for we has provided the limited scopes.
In our cases， we create two resource quota in the same namespace,  and we want to check the "used" quota for different workload.

**Comment by yliaog** [comment]
could you be more specific about what resource quota (object count, or cpu/ram) are you using for workload? example is even better to help me understand your use case.

**Comment by ddongchen** [comment]
The resource quota we care about is cpu/ram.

For example, we create an object via "CRD", which could also create pods. and we want to create the resourcequota for the object,  and the quota should not be influenced by original pods.

**Comment by ddongchen** [comment]
we created an object via "CRD", and we wrote a controller for it.

**Comment by yliaog** [comment]
could you separate them into their own namespace? if not, why not?

For the workload in one namespace, they likely are managed by the same team, have the same access, they usually cooperate to share the quota allocated to the namespace. If you want to specify very specific custom policy, probably you can write a custom policy controller of your own, that monitors the resource usages, and takes proper actions when the policies are violated.

**Comment by ddongchen** [comment]
In the same namespace, the pods may be created by deployment, or created by our own "CRD" controller. we just want to classify it via different resource quota.
